### PR TITLE
docs: add more information regarding --use-new-run

### LINKED ISSUE #2575: document more clearly what the purpose of the --use-new-run feature is and how it works
How does the feature --use-new-run work?

I will be happy to explain :)
I didn't find information in the documentation.

### PR DESCRIPTION
**Pull request by aaron-prindle**
fixes #2575

### CODE REVIEW

**Review by JeromeJu — APPROVED** [review]
Thanks @aaron-prindle 
lgtm

File: README.md [hunk 1]
```diff
@@ -1068,8 +1068,12 @@ Set this flag to indicate which build stage is the target build stage.
 
 #### Flag `--use-new-run`
 
-Use the experimental run implementation for detecting changes without requiring
-file system snapshots. In some cases, this may improve build performance by 75%.
+Use the experimental RUN implementation for detecting changes without requiring
```
Comment (braykov) [inline README.md:1071]: This sentence is now broken and incomplete.

Comment (braykov) [reply README.md:1071]: Not sure if anyone gets notified on PR comments since this is merged, so: @aaron-prindle

Comment (aaron-prindle) [reply README.md:1071]: Thanks for flagging this @braykov, submitted https://github.com/GoogleContainerTools/kaniko/pull/2698 to fix this

**Pull request merged**